### PR TITLE
[SNOW-1357748]: Update `read_snowflake` to take into account new logical plan format

### PR DESCRIPTION
**Pull request by sfc-gh-rdurrani**
Please answer these questions before submitting your pull requests. Thanks!

1. What GitHub issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   Fixes SNOW-1357748

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency

3. Please describe how your code solves the related issue.
Fixes read_snowflake.

### CODE REVIEW

**Review by sfc-gh-azhan — APPROVED** [review]
Let's merge after all prod is on 8.17

File: src/snowflake/snowpark/modin/plugin/_internal/utils.py [hunk 1]
```diff
@@ -166,26 +166,25 @@ def _check_if_sql_query_contains_order_by_and_warn_user(
...
-    # 1   1.0  0.0            None         Result ...
-    # 2   1.0  1.0             [0]  SortWithLimit ...
-    # 3   1.0  2.0             [1]      TableScan ...
-    potential_sort_row = logical_plan[
-        (logical_plan["parentOperators"] == "[0]")
-        & (logical_plan["operation"].str.lower().str.contains("sort"))
-    ]
-    contains_order_by = len(potential_sort_row) > 0
+    # Have to use `strip` since the names of the operators contain indentation.
+    contains_order_by = (logical_plan["operation"].str.strip() == "Sort").any()
```
Comment (sfc-gh-nkumar) [inline src/snowflake/snowpark/modin/plugin/_internal/utils.py:187]: Was this is a BCR change or just internal change?

Comment (sfc-gh-rdurrani) [reply src/snowflake/snowpark/modin/plugin/_internal/utils.py:187]: Hmm - I'm not actually sure - explain is a public operator, so the output from explain changing could be a BCR?

**Review by sfc-gh-mvashishtha — COMMENTED** [review]
@sfc-gh-rdurrani please un-skip all the tests marked with `@pytest.mark.skip(reason="SNOW-1358681")` in 

- test_read_snowflake_query_cte.py
- test_read_snowflake_query_order_by.py
- test_read_snowflake_select_query.py

File: src/snowflake/snowpark/modin/plugin/_internal/utils.py (hunk 1 above)
Comment (sfc-gh-azhan) [reply src/snowflake/snowpark/modin/plugin/_internal/utils.py:187]: https://snowflake.slack.com/archives/C02BTC3HY/p1714594898428329

**Review by sfc-gh-azhan — APPROVED** [review]

**Review by sfc-gh-mvashishtha — APPROVED** [review]
LGTM, thanks!

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM!

**Pull request merged**